In [1]:
import pandas as pd
import numpy as np

In [12]:
data = pd.read_csv('bank_data.csv')
data.dtypes

age           int64
job          object
marital      object
education    object
default      object
balance       int64
housing      object
loan         object
contact      object
day           int64
month        object
duration      int64
campaign      int64
pdays         int64
previous      int64
poutcome     object
deposit      object
dtype: object

### 数据完整性

In [10]:
### 字段缺失情况
data.count()/len(data)

age          1.0
job          1.0
marital      1.0
education    1.0
default      1.0
balance      1.0
housing      1.0
loan         1.0
contact      1.0
day          1.0
month        1.0
duration     1.0
campaign     1.0
pdays        1.0
previous     1.0
poutcome     1.0
deposit      1.0
dtype: float64

- 各字段均不存在缺失情况

### 数据统计

In [11]:
data.describe()

age       balance           day      duration      campaign  \
count  11162.000000  11162.000000  11162.000000  11162.000000  11162.000000   
mean      41.231948   1528.538524     15.658036    371.993818      2.508421   
std       11.913369   3225.413326      8.420740    347.128386      2.722077   
min       18.000000  -6847.000000      1.000000      2.000000      1.000000   
25%       32.000000    122.000000      8.000000    138.000000      1.000000   
50%       39.000000    550.000000     15.000000    255.000000      2.000000   
75%       49.000000   1708.000000     22.000000    496.000000      3.000000   
max       95.000000  81204.000000     31.000000   3881.000000     63.000000   

              pdays      previous  
count  11162.000000  11162.000000  
mean      51.330407      0.832557  
std      108.758282      2.292007  
min       -1.000000      0.000000  
25%       -1.000000      0.000000  
50%       -1.000000      0.000000  
75%       20.750000      1.000000  
max      854.000000     58.000000

以上为数值型变量的简单统计分析
- 平均年龄41岁
- 账户余额均值1529
- 平均参加营销活动次数2.5次

### 数据关联
简单分析年龄、职业、婚姻状况、受教育程度四个变量与账户余额的关系

In [16]:
# 年龄与账户余额的相关性
data[['age','balance']].corr()

age  balance
age      1.0000   0.1123
balance  0.1123   1.0000

In [21]:
# 按职业分组的账户余额均值
data.groupby('job')['balance'].mean().sort_values(ascending=False)

job
retired          2417.250643
unknown          1945.457143
self-employed    1865.372840
management       1793.663679
entrepreneur     1621.942073
technician       1556.294569
student          1500.783333
housemaid        1366.160584
unemployed       1314.719888
blue-collar      1203.926440
admin.           1195.866567
services         1081.171181
Name: balance, dtype: float64

In [22]:
# 按婚姻状况分组的账户余额均值
data.groupby('marital')['balance'].mean().sort_values(ascending=False)

marital
married     1599.927570
single      1457.255259
divorced    1371.835267
Name: balance, dtype: float64

In [23]:
# 按受教育程度分组的账户余额均值
data.groupby('education')['balance'].mean().sort_values(ascending=False)

education
tertiary     1845.869070
unknown      1746.605634
primary      1523.031333
secondary    1296.480278
Name: balance, dtype: float64

以上分析显示：
- 年龄与账户资产的相关系数为0.11，正向弱相关；
- 已退休客户账户平均资产最高，其次为自雇者和管理人员；
- 已婚客户账户平均资产最高，其次为单身客户和离婚客户；
- 受过高等教育客户的账户平均资产最高，但受过初等教育客户的平均资产高于中等教育客户，可能源于数据代表性的问题。